magma/verilog_utils.py

```diff
@@ -8,43 +8,57 @@
 from magma.view import InstView, PortView
 
 
-def value_to_verilog_name(value):
-    if isinstance(value, Array) and not issubclass(value.T, Digital):
-        elems = ", ".join(value_to_verilog_name(t) for t in reversed(value))
+def is_nd_array(T, skip=True):
```

Comment:
what is the meaning of skip? Looks like for the top level call we're allowed to return true if digital? So the top level call for `is_nd_array(Bit)` returns true? Why is this desired?

Reply:
Actually it's the other way around `is_nd_array(Bit)` will return False (since skip defaults to True), so we require at least one level of array before we a see a Bit and return True.